pkg/network/openshift_sdn.go

```diff
@@ -48,10 +49,28 @@ func renderOpenShiftSDN(conf *operv1.NetworkSpec, manifestDir string) ([]*uns.Un
 		"metrics-bind-address":    {"0.0.0.0"},
 		"metrics-port":            {"9101"},
 		"healthz-port":            {"10256"},
-		"proxy-mode":              {"iptables"},
+		"proxy-mode":              {"unidling+iptables"},
 		"iptables-masquerade-bit": {"0"},
 	}
-	kpc, err := kubeProxyConfiguration(conf, kpcDefaults)
+	if !*c.EnableUnidling {
+		kpcDefaults["proxy-mode"][0] = "iptables"
+	}
+
+	// If the user specified a proxy-mode, append "unidling+" to it, since
+	// that's an internal detail that was added in 4.2.
+	kpcOverrides := map[string]operv1.ProxyArgumentList{}
+	if *c.EnableUnidling && len(conf.KubeProxyConfig.ProxyArguments["proxy-mode"]) > 0 {
+		pma := conf.KubeProxyConfig.ProxyArguments["proxy-mode"]
+		if !strings.HasPrefix(pma[0], "unidling+") {
```

Comment:
FWIW, validation has already confirmed that if `*c.EnableUnidling` is set (after defaulting), then `conf.KubeProxyConfig.ProxyArguments["proxy-mode"]` is either unset or `["iptables"]`. So you can simplify this to just:

    kpcOverrides := map[string]operv1.ProxyArgumentList{}
    if *c.EnableUnidling {
            kpcOverrides["proxy-mode"] = operv1.ProxyArgumentList{"unidling+iptables"}`
    }

Reply:
Ah yeah, good point.